### PR TITLE
Updating DOI code to utilize prefix environment variables

### PR DESCRIPTION
**Pull request by carolyncole**
Updated the code to use a variable DOI prefix based on the environment. 

Updated error handling in DOI generation and when fetching S3 files associated with a DOI that has no files in S3 yet. Also added special condition to allow the system to work in development without having to have the DataCite credentials.

### CODE REVIEW

**Comment by hectorcorrea** [comment]
@carolyncole Thank you so much for this. I made a few changes to the PR so that it handles errors properly (the fancy `.either` syntax was not really working) and added special clause to work in development without credentials. I deployed it to staging and it works OK.

File: app/services/s3_query_service.rb [hunk 1]
```diff
@@ -45,10 +45,9 @@ def client
   # * https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/S3/Client.html#get_object_attributes-instance_method
   # @return [<S3File>] An Array of S3File objects
   def data_profile
-    return [] if @doi == "10.34770/tbd"
     objects = []
     resp = client.list_objects_v2({ bucket: bucket_name, max_keys: 1000, prefix: prefix })
-    resp.to_h[:contents].each do |object|
+    resp.to_h[:contents]&.each do |object|
```
Comment (hectorcorrea) [inline app/services/s3_query_service.rb:50]: Prevents this code from crashing if the S3 bucket does not exist.

File: app/models/work.rb [hunk 2]
```diff
@@ -122,15 +122,17 @@ def default_work(title, user_id, collection_id, datacite_resource, ark)
...
   def draft_doi
-    self.doi ||= "10.34770/tbd"
-    # TODO: Set up the doi to have  a variable prefix.  Test and production do not have the same one
-    # self.doi ||= begin
-    #                result = data_cite_connection.autogenerate_doi(prefix: "10.34770")
-    #                result.either(
-    #                   ->(response) { response.doi },
-    #                   ->(response) { raise("Something went wrong", response.status) }
-    #                 )
-    #              end
```
Comment (hectorcorrea) [inline app/models/work.rb:133]: The `either` syntax is what the datacite gem recommends (https://github.com/sul-dlss/datacite-ruby) is more elegant but does not seem to  really catch errors so I ditched it in favor of a good and tried if/else.

File: app/models/work.rb [hunk 3]
```diff
@@ -122,15 +122,17 @@ def default_work(title, user_id, collection_id, datacite_resource, ark)
...
+    self.doi ||= if Rails.env.development? && ENV["DATACITE_USER"].blank?
+                   Rails.logger.info "Using hard-coded test DOI during development."
+                   "10.34770/tbd"
```
Comment (jrgriffiniii) [inline app/models/work.rb:127]: 👍

**Review by jrgriffiniii — APPROVED** [review]

**Comment by jrgriffiniii** [comment]
Coveralls does not show any decrease in the test coverage (however, the API seems to be slow)

File: README.md [hunk 4]
```diff
@@ -45,6 +45,9 @@ This is convenient when you need to step through the code as `byebug` does not a
 2. Terminal two: `bin/webpack-dev-server`
 3. Access pdc_describe at [http://localhost:3000/](http://localhost:3000/)
 
+## DataCite integration
```
Comment (jrgriffiniii) [inline README.md:48]: 💯

**Review by jrgriffiniii — APPROVED** [review]

**Pull request merged**